runtime/drivers/mysql/sql_store.go

```diff
@@ -0,0 +1,216 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	sqldriver "database/sql/driver"
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/mitchellh/mapstructure"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+// Query implements drivers.SQLStore
+func (c *connection) Query(ctx context.Context, props map[string]any) (drivers.RowIterator, error) {
+	srcProps, err := parseSourceProperties(props)
+	if err != nil {
+		return nil, err
+	}
+
+	var dsn string
+	if srcProps.DSN != "" { // get from src properties
+		dsn = srcProps.DSN
+	} else if url, ok := c.config["dsn"].(string); ok && url != "" { // get from driver configs
+		dsn = url
+	} else {
+		return nil, fmt.Errorf("the property 'dsn' is required for MySQL. Provide 'dsn' in the YAML properties or pass '--env connector.mysql.dsn=...' to 'rill start'")
+	}
+
+	conf, err := mysql.ParseDSN(dsn)
+	if err != nil {
+		return nil, err
+	}
+	conf.ParseTime = true // if set to false, time is scanned as an array rather than as time.Time
+
+	db, err := sql.Open("mysql", conf.FormatDSN())
+	if err != nil {
+		db.Close()
```

Comment:
I don't think `db.Close` should be called if `Open` fails (`db` will be `nil` in that case and may give nil access panic)